contracts/internal/EverlongERC4626.sol

```diff
@@ -54,6 +34,97 @@ abstract contract EverlongERC4626 is ERC4626 {
         _decimals = success ? result : _DEFAULT_UNDERLYING_DECIMALS;
     }
 
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Stateful                                                │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @inheritdoc ERC4626
+    function deposit(
+        uint256 assets,
+        address to
+    ) public override returns (uint256 shares) {
+        IERC20(_asset).approve(hyperdrive, assets);
+        shares = super.deposit(assets, to);
+    }
+
+    /// @inheritdoc ERC4626
+    function redeem(
```

Comment:
A question that I'd have about this function is whether or not we're going to include withdrawal shares in this implementation. I think that it would really hurt the UX not to have them.